### ISSUE TITLE
Problem with babel-loader -> Wallaby.js does not work in most of my files.

### ISSUE THREAD

**Issue by aurelienbottazini** [wallabyjs/public#1486]
### Issue description or question
I am trying to use Wallaby with Vue, Webpack, Mocha

It seems there is a loading error with babel 

This is one for the many error messages I get:
<img width="1119" alt="doximity-client-vue____projects_work_doximity-client-vue__-_____src_packages_colleagues_components_colleaguebutton_vue__doximity-client-vue_" src="https://user-images.githubusercontent.com/32635/35454039-b44aaae8-02cc-11e8-8d7e-13516165d40d.png">

This the failing line inside babel-loader.
<img width="534" alt="image" src="https://user-images.githubusercontent.com/32635/35454337-ed86f4be-02cd-11e8-8ad1-9b02e8ef42f4.png">

WallabyJS does work with some of my files and for others it does not and just display that the file is not covered. I think it is because of this babel loader error.

https://github.com/ChangJoo-Park/vue-wallaby-webpack-template works fine for me.
I have another react-native project where wallaby.js works fine too

I do have a .babelrc file at the root of my project.

### Wallaby.js configuration file

```javascript
// paste your config here
const webpackConfig = require('./build/webpack.test.config');
const wallabyWebpack = require('wallaby-webpack');
const { resolvePath } = require('./build/utils');

// process.env.BABEL_ENV = 'test';

module.exports = function(wallaby) {
  webpackConfig.resolve.alias = {
    '@': require('path').join(wallaby.projectCacheDir, 'src'),
    vue: 'vue/dist/vue.min.js',
    '~src': resolvePath('src'),
    '~server': resolvePath('server'),
    '~test': resolvePath('test'),
    '~lib': resolvePath('src/lib'),
    '~public': resolvePath('public'),
    '~packages': resolvePath('src/packages'),
    request$: 'xhr',
    '~mock': resolvePath('__mock__'),
  };
  webpackConfig.externals = { vue: 'Vue' };
  webpackConfig.module.rules.find(
    r => r.loader === 'vue-loader'
  ).options.loaders.js =
    '';
  webpackConfig.plugins.push(
    new (require('webpack')).LoaderOptionsPlugin({
      test: /\.vue$/,
      sourceMap: false,
    })
  );

  const wallabyPostprocessor = wallabyWebpack(webpackConfig);

  return {
    files: [
      { pattern: '.babelrc', instrument: false },
      {
        pattern: 'node_modules/babel-polyfill/dist/polyfill.js',
        instrument: false,
      },
      { pattern: 'node_modules/vue/dist/vue.js', instrument: false },

      { pattern: 'node_modules/sinon/pkg/sinon.js', instrument: false },
      { pattern: 'node_modules/chai/chai.js', instrument: false },
      {
        pattern: 'node_modules/sinon-chai/lib/sinon-chai.js',
        instrument: false,
      },
      { pattern: 'node_modules/chai-string/chai-string.js', instrument: false },
      { pattern: 'node_modules/chai-subset/chai-subset.js', instrument: false },
      { pattern: 'src/**/*.*', load: false },
      { pattern: 'src/**/*.spec.js', ignore: true },
      { pattern: 'src/**/.DS_Store', ignore: true },
      { pattern: 'src/dox_assets/**/*.*', ignore: true },
    ],

    env: {
      type: 'browser',
    },

    tests: [{ pattern: 'src/**/*.spec.js', load: false }],

    postprocessor: wallabyPostprocessor,

    setup() {
      Vue.config.errorHandler = function(err) {
        throw err;
      };
      window.expect = chai.expect;
      var should = chai.should();
      window.__moduleBundler.loadTests();
    },

    hints: {
      ignoreCoverage: /ignore coverage/,
    },
    compilers: {
      '**/*.js': wallaby.compilers.babel(),
      '**/*.vue': require('wallaby-vue-compiler')(wallaby.compilers.babel()),
    },
    debug: true,

    testFramework: 'mocha',
  };
};
```

and my webpack test config
```
const merge = require('webpack-merge');
const webpack = require('webpack');
const baseConfig = require('./webpack.base.config.js');
const { clientConfig: { MAPBOX_TOKEN } } = require('../lib/config');

const webpackConfig = merge(baseConfig, {
  module: {},
  devtool: 'cheap-module-eval-source-map',
  resolve: {
    alias: {
      vue: 'vue/dist/vue.min.js',
    },
  },
});

webpackConfig.plugins = [
  new webpack.DefinePlugin({
    'process.env': {
      NODE_ENV: JSON.stringify('test'),
      UNIT_TEST: JSON.stringify('true'),
      SKIP_OAUTH: JSON.stringify('true'),
      CLOUDINARY_NAME: JSON.stringify('testcloud'),
      DOXIMITY_URL: JSON.stringify('http://mydox.com'),
      EMBEDLY_API_KEY: JSON.stringify('embedly-api-key'),
      // sauce lab vms are slow!
      TRANSITION_DURATION: JSON.stringify(500),
      TRANSITION_BUFFER: JSON.stringify(50),
      GRAPHQL_URL: JSON.stringify('http://localhost:5000/graphql'),
      MAPBOX_TOKEN: JSON.stringify(MAPBOX_TOKEN),
      CIRCLE_BRANCH: JSON.stringify(process.env['CIRCLE_BRANCH']),
    },
  }),
  new webpack.NormalModuleReplacementPlugin(
    /\.(eot|woff|woff2|ttf|svg|png|jpe?g|gif)(\?\S*)?$/,
    'node-noop'
  ),
];

webpackConfig.module.rules = [
  {
    test: /\.vue$/,
    loader: 'vue-loader',
    options: {
      loaders: {
        scss: 'null-loader',
        sass: 'null-loader',
        css: 'null-loader',
      },
    },
  },
  {
    test: /\.js$/,
    loader: 'babel-loader',
    exclude: /node_modules/,
  },
  {
    test: /\.(sass|scss|css)$/,
    use: [
      {
        loader: 'style-loader/useable',
      },
      {
        loader: 'null-loader',
      },
    ],
  },
];

delete webpack.entry;

module.exports = webpackConfig;
```

### Code editor or IDE name and version

IntelliJ IDEA 2017.3.3 (Ultimate Edition)
Build #IU-173.4301.25, built on January 15, 2018

### OS name and version
OSX 10.13.2

**Comment by aurelienbottazini** [comment]
Those for your work, I bought a licence for Wallaby.js and Quokka. Your tools are awesome!

**Comment by ArtemGovorov** [comment]
While I'm not sure what is causing the issue, wallaby doesn't actually need `babel-loader`, so this should help:
```diff

...

  webpackConfig.module.rules.find(
    r => r.loader === 'vue-loader'
  ).options.loaders.js =
    '';
+ webpackConfig.module.rules = webpackConfig.module.rules.filter(r=> r.loader !== 'babel-loader')  
  webpackConfig.plugins.push(
    new (require('webpack')).LoaderOptionsPlugin({
      test: /\.vue$/,
      sourceMap: false,
    })
  );

...
```

Also these lines may cause wallaby not work as you'd expect:

```javascript
    '~src': resolvePath('src'),
    '~server': resolvePath('server'),
    '~test': resolvePath('test'),
    '~lib': resolvePath('src/lib'),
    '~public': resolvePath('public'),
    '~packages': resolvePath('src/packages'),
    ...
    '~mock': resolvePath('__mock__'),
```

If the `resolvePath` function resolves your local project folder path, then when running wallaby these alias/files may not have coverage and may not work. If that's the case, you'll need to specify paths for them just like you're doing it for the `@` alias (pointing to the `wallaby.projectCacheDir` folder):
```
'@': require('path').join(wallaby.projectCacheDir, 'src'),
```

**Comment by aurelienbottazini** [comment]
Ok, i think I am getting a little farther but I am still stuck.
I decided to test file by file and see what's working and what's failing. 

Here is my new webpack config:
```
const webpackConfig = require('./build/webpack.test.config');
const wallabyWebpack = require('wallaby-webpack');
const { resolvePath } = require('./build/utils');

// process.env.BABEL_ENV = 'test';

module.exports = function(wallaby) {
  webpackConfig.resolve.alias = {
    vue: 'vue/dist/vue.min.js',
    '~src': resolvePath('src'),
    '~server': resolvePath('server'),
    '~test': resolvePath('test'),
    '~lib': resolvePath('src/lib'),
    '~public': resolvePath('public'),
    '~packages': resolvePath('src/packages'),
    request$: 'xhr',
    '~mock': resolvePath('__mock__'),
  };
  webpackConfig.externals = { vue: 'Vue' };
  webpackConfig.entry = './src/app.js';
  webpackConfig.module.rules.find(
    r => r.loader === 'vue-loader'
  ).options.loaders.js =
    '';
  webpackConfig.module.rules = webpackConfig.module.rules.filter(
    r => r.loader !== 'babel-loader'
  );
  webpackConfig.plugins.push(
    new (require('webpack')).LoaderOptionsPlugin({
      test: /\.vue$/,
      sourceMap: false,
    })
  );

  const wallabyPostprocessor = wallabyWebpack(webpackConfig);

  return {
    files: [
      {
        pattern: 'node_modules/babel-polyfill/dist/polyfill.js',
        instrument: false,
      },
      { pattern: 'node_modules/vue/dist/vue.js', instrument: false },

      { pattern: 'node_modules/sinon/pkg/sinon.js', instrument: false },
      { pattern: 'node_modules/chai/chai.js', instrument: false },
      {
        pattern: 'node_modules/sinon-chai/lib/sinon-chai.js',
        instrument: false,
      },
      { pattern: 'node_modules/chai-string/chai-string.js', instrument: false },
      { pattern: 'node_modules/chai-subset/chai-subset.js', instrument: false },
      { pattern: 'src/dox_assets/**/*.*', ignore: true },
      { pattern: 'src/**/__tests__/*.*', ignore: true },
      { pattern: 'src/**/*.*', load: false },
    ],

    env: {
      type: 'browser',
    },

    tests: [
      {
        pattern: 'src/packages/base/components/__tests__/Modal*.spec.js',
        load: false,
      },
    ],

    postprocessor: wallabyPostprocessor,

    setup() {
      Vue.config.errorHandler = function(err) {
        throw err;
      };
      window.expect = chai.expect;
      var should = chai.should();
      window.__moduleBundler.loadTests();
    },

    hints: {
      ignoreCoverage: /ignore coverage/,
    },
    compilers: {
      '**/*.js': wallaby.compilers.babel({
        presets: ['es2015', 'es2016', 'es2017'],
        plugins: [
          'transform-runtime',
          'transform-es2015-modules-commonjs',
          'syntax-dynamic-import',
          'transform-object-rest-spread',
          'dynamic-import-node',
        ],
      }),
      '**/*.vue': require('wallaby-vue-compiler')(
        wallaby.compilers.babel({
          presets: ['es2015', 'es2016', 'es2017'],
          plugins: [
            'transform-runtime',
            'transform-es2015-modules-commonjs',
            'syntax-dynamic-import',
            'transform-object-rest-spread',
            'dynamic-import-node',
          ],
        })
      ),
    },
    debug: true,

    testFramework: 'mocha',
  };
};
```

I removed the babel-loader from webpack, I am ignoring the js files and I am just testing one file.
It works with another file but not with this one because it seems the spread plugin for objects is not loaded.

<img width="690" alt="doximity-client-vue____projects_work_doximity-client-vue__-_____wallaby_js__doximity-client-vue_" src="https://user-images.githubusercontent.com/32635/35876192-c78a9700-0b71-11e8-8c26-89af508b2a88.png">


It fails because of the three **...**
I have a babelrc file but i inlined the configuration to try to debug but it seems even inlined this is not working.
I think something is wrong with the babel setup but I am not sure what.

**Comment by aurelienbottazini** [comment]
Thanks a lot for your help @ArtemGovorov

**Comment by ArtemGovorov** [comment]
I think `webpackConfig.resolve.alias` pointing to your local project is causing the issue. As I have mentioned:

> Also these lines may cause wallaby not work as you'd expect:

```javascript
    '~src': resolvePath('src'),
    '~server': resolvePath('server'),
    '~test': resolvePath('test'),
    '~lib': resolvePath('src/lib'),
    '~public': resolvePath('public'),
    '~packages': resolvePath('src/packages'),
    ...
    '~mock': resolvePath('__mock__'),
```

> If the `resolvePath` function resolves your local project folder path, then when running wallaby these alias/files may not have coverage and may not work. If that's the case, you'll need to specify paths for them just like you're doing it for the `@` alias (pointing to the `wallaby.projectCacheDir` folder):
```
'@': require('path').join(wallaby.projectCacheDir, 'src'),
```

So please try pointing all of the alias paths to the wallaby cache, ie.

```javascript
    '~src': require('path').join(wallaby.projectCacheDir, 'src'),
    '~server': require('path').join(wallaby.projectCacheDir, 'server'),
    // etc
```

**Comment by aurelienbottazini** [comment]
Thanks a lot following your advice and with some more tweaking (loading more files) I was able to solve this babel error

**Comment by ArtemGovorov** [comment]
Awesome, thanks for the update!

**Comment by maksimr** [comment]
We have the same problem after moving on `webpack@4` and `babel-loader@7.1.5`

```
    "babel-loader": "6.4.1",
    "babel-loader": "7.1.5",
```


```
Uncaught Error: Module build failed (from ./node_modules/ring-hub/node_modules/babel-loader/lib/index.js):
TypeError: Cannot read property 'toString' of undefined
    at readBabelConfig
```

If add console.log to `read.js` like this
```js
  if (fileSystem.readFileSync(filename) === undefined) {
    console.log(filename, fileSystem.readFileSync(filename));
  }
```

Output:

```
node_modules/ring-hub/.babelrc undefined
```

File `node_modules/ring-hub/.babelrc ` exists
Karma tests also work fine.

Maybe wallaby stub or tweak `fileSystem` object?
Because if replace `fileSystem.readFileSync(filename)` on `require('fs').readFileSync(filename)` it returns  .babelrc's content

**Comment by maksimr** [comment]
Looks like the problem in `wallaby-webpack`

`readFileSync` uses `fs.readFile.apply(fs, arguments);` instead of `fs.readFileSync.apply(fs, arguments);`
https://github.com/jeffling/wallaby-webpack/blob/master/lib/WallabyInputFileSystem.js#L51-L60

**Comment by ArtemGovorov** [comment]
@maksimr Awesome, thanks for investigating and sending the pull request. It's now merged and published in `wallaby-webpack` v3.9.15.

**Comment by maksimr** [comment]
@ArtemGovorov no problem! New version works perfectly
